### PR TITLE
ci: github workflow

### PR DESCRIPTION
**Pull request by saihaj**
### Summary of PR
* [X] Add GitHub Workflow for CI
* [ ]  Release workflow for publishing
### Tests for unexpected behavior
You can test run ->  https://github.com/saihaj/ShabadOSdesktop/actions/runs/144226934 

image

### Time spent on PR
2 hours

### Reviewers
@Harjot1Singh

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - name: Run ESLint on Frontend & Backend
+        run: npm run lint
+
+  build:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    needs: lint
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (Harjot1Singh) [inline .github/workflows/ci.yml:49]: Nice one!

**Comment by saihaj** [comment]
Can;t really remove CCI, since we need it for releases

**Comment by saihaj** [comment]
@Harjot1Singh we have 2 options to choose from when creating and publishing release in gihtub actions:

1. We create a release using [`actions/create-release@v1`](https://github.com/actions/create-release) and generate changelog (You can see example of this here -> https://github.com/saihaj/ShabadOSdesktop/runs/973714966). Then we have another workflow in which we create binaries for OS and then using this [`bruceadams/get-release@v1.2.0`](bruceadams/get-release@v1.2.0) we get the draft release we created and upload assets to it and finally update the release to mark it published using something like [ `johnwbyrd/update-release@v1.0.0`](https://github.com/marketplace/actions/update-release)

2. Other option is that will allows us to keep one file. We extend on what I have right now for CI once the binaries are built we upload them to GitHub actions artifacts section(similar to CCI job) using [`actions/upload-artifact@v2`](https://github.com/actions/upload-artifact) and then have a new job that will depend on build step (once all the binaries have been built and uploaded to gh-actions artifacts section) we create a release using [`actions/create-release@v1`](https://github.com/actions/create-release), retrieve built binaries using [`actions/download-artifact@v2`](https://github.com/actions/upload-artifact) and upload them to our release using [`actions/upload-release-asset@v1`](https://github.com/actions/upload-release-asset)

**Comment by bhajneet** [comment]
Please rename the title of the PR to match our [commit guidelines](https://github.com/ShabadOS/.github/wiki/How-to-Contribute#commit-messages)

I think in this case the type would be "build" instead of "ci"

**Comment by bhajneet** [comment]
That is my bad, I honestly didn't know we had a "ci" type -- My apologies!

**Comment by Harjot1Singh** [comment]
@saihaj what's the difference to what comes out, between the two?

**Comment by bhajneet** [comment]
@saihaj are you planning on working on this? if not, close the PR. If it's something you want to work on but can't because you're waiting on something else change the status from Hold to "Blocked" so it's in the correct PM column.

**Comment by saihaj** [comment]
Now we have the actions repo we will most likely use those. I guess it is easier to just make a new one

**Pull request closed without merging**